### PR TITLE
Updates to grades page.

### PR DESCRIPTION
**Pull request by somiaj**
Make test names link to the ProblemSet test page. If it is
  a test version, (version n) links to the actual test version,
  provided the student can see the version's problems. Also
  consistently use table headers for the first column.

  For sets that are not yet opened, don't link to the set and
  don't show the grades for the set.

  Don't show hidden sets on the grades page.

  Refer to tests as tests (instead of set or assignment) in messages
  about no version has been taken and cannot display scores.

### CODE REVIEW

**Comment by somiaj** [comment]
It should be noted that this also changes what is shown on the 'Student Progress' page for individual students, since that is identical to what you would see on the grades page for that student.

File: lib/WeBWorK/ContentGenerator/Grades.pm [hunk 1]
```diff
@@ -202,7 +205,8 @@ sub displayStudentStats ($c, $studentID) {
 			# For other sets we just count the number of problems.
 			$num_of_problems = $db->countGlobalProblems($setID);
 		}
-		$max_problems = $max_problems < $num_of_problems ? $num_of_problems : $max_problems;
+		$max_problems =
+			(($set && after($set->open_date)) && $max_problems < $num_of_problems) ? $num_of_problems : $max_problems;
```
Comment (drgrice1) [inline lib/WeBWorK/ContentGenerator/Grades.pm:209]: There are extra parentheses here that aren't needed.  The only necessary parentheses are the ones around the argument of the `after` method.

File: lib/WeBWorK/ContentGenerator/Grades.pm [hunk 2]
```diff
@@ -215,22 +219,56 @@ sub displayStudentStats ($c, $studentID) {
...
-							colspan => $max_problems + 3,
-							$c->tag('em', $c->maketext('No versions of this assignment have been taken.'))
+							'em',
+							after($set->open_date) ? $c->maketext('No versions of this test have been taken.')
+							: $c->maketext(
+								'Will open on [_1].',
+								$c->formatDateTime($set->open_date, undef, $ce->{studentDateDsiplayFormat})
+							)
 						)
+
```
Comment (drgrice1) [inline lib/WeBWorK/ContentGenerator/Grades.pm:268]: There is an empty line here that shouldn't be.

**Review by drgrice1 — APPROVED** [review]
This looks good.  Just a few quibbles on code style.

**Pull request merged**